rust-bitcoin-derive/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+//! Custom derive macros for the rust-bitcoin project.
+//!
+
+extern crate proc_macro;
+extern crate syn;
+extern crate quote;
+
+use proc_macro::TokenStream;
+use syn::{parse_macro_input, Data, DeriveInput, Fields};
+use quote::quote;
+
+#[proc_macro_derive(Encodable)]
+pub fn encodable_derive(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+
+    let ident = input.ident;
+    let data = input.data;
+
+    let fields = match data {
+        Data::Struct(ref data) => {
+            match data.fields {
+                Fields::Named(_) => &data.fields,
+                Fields::Unnamed(_) | Fields::Unit => {
+                    panic!("Encodable only implemented for named structs");
```

Comment:
I think the right way is to create a compile_error with the appropriate span, `syn` helps you do that via: https://docs.rs/syn/latest/syn/parse/struct.Error.html#